### PR TITLE
chore: add sonar cache

### PR DESCRIPTION
**Pull request by viniciusps**

### CODE REVIEW

File: code-quality/action.yaml [hunk 1]
```diff
@@ -31,6 +31,14 @@ runs:
         name: coverage
         path: ${{ inputs.coverage_dir }}
 
+    - uses: actions/cache@v2
+      id: sonar-cache
+      with:
+        path: /opt/sonar-scanner/.sonar/cache
+        key: ${{ runner.os }}-sonar
```
Comment (JoshVee) [inline code-quality/action.yaml:38]: i don't think `runner.os` is specific enough. Maybe use branch name also

Comment (viniciusps) [reply code-quality/action.yaml]: yep. Fixed

File: code-quality/action.yaml [hunk 2]
```diff
@@ -31,6 +31,14 @@ runs:
         name: coverage
         path: ${{ inputs.coverage_dir }}
 
+    - uses: actions/cache@v2
+      id: sonar-cache
+      with:
+        path: /opt/sonar-scanner/.sonar/cache
+        key: ${{ runner.os }}-${{ env.CI_REPOSITORY_NAME_SLUG }}-sonar
```
Comment (JoshVee) [inline code-quality/action.yaml:38]: branch name

Comment (JoshVee) [reply code-quality/action.yaml]:
```suggestion
        key: ${{ runner.os }}-sonar-${{ hashFiles('**/sonar-project.properties') }}
```

File: code-quality/action.yaml [hunk 3]
```diff
@@ -31,6 +31,14 @@ runs:
...
+        restore-keys: |
+          ${{ runner.os }}-${{ env.CI_REPOSITORY_NAME_SLUG }}
```
Comment (JoshVee) [inline code-quality/action.yaml]:
```suggestion
          ${{ runner.os }}-sonar-
```

File: code-quality/action.yaml [hunk 4]
```diff
@@ -31,6 +31,14 @@ runs:
         name: coverage
         path: ${{ inputs.coverage_dir }}
 
+    - uses: actions/cache@v2
+      id: sonar-cache
+      with:
+        path: /opt/sonar-scanner/.sonar/cache
+        key: ${{ runner.os }}-${{ env.CI_HIPAGES_BRANCH_SLUG }}-sonar
+        restore-keys: |
+          ${{ runner.os }}-${{ env.CI_HIPAGES_BRANCH_SLUG }}
```
Comment (JoshVee) [inline code-quality/action.yaml:40]:
```suggestion
        key: ${{ runner.os }}-sonar-${{ hashFiles('**/sonar-project.properties') }}
        restore-keys: |
          ${{ runner.os }}-sonar-
```

**Review by JoshVee — APPROVED** [review]

**Pull request merged**